src/getDatasetHelpers.js

```diff
@@ -56,6 +57,8 @@ const splitPrefixIntoParts = (prefix) => {
   }
 
   const Source = sources.get(sourceName);
+  // Source not found error
+  if (typeof Source !== "function") throw new ResourceNotFoundError();
```

Comment:
I think it's valid to use `ResourceNotFoundError` in this case, as we have examples where this is used internally to end up with the client being sent a `404`. 

Here we're within `splitPrefixIntoParts` which is called from ~7 different locations. Each of these should be wrapped in a `try/catch` block, including a specific handler for `ResourceNotFoundError` where appropriate. Looking at the code this happens in only one or two cases!